xload/async.go

```diff
@@ -0,0 +1,182 @@
+package xload
+
+import (
+	"context"
+	"reflect"
+	"sync"
+)
+
+// LoadAsync loads values concurrently by calling Loader in parallel.
+// Number of goroutines can be controlled with xload.Concurrency.
+func LoadAsync(ctx context.Context, v any, opts ...Option) error {
```

Comment:
What if we use an `Option` to control sync/async processing instead of exposing a new top level function?

Example: `xload.Concurrency(3) > 0 ? processAsync : process`

Reply:
Agreed. But keeping the default as 1